src/main/java/io/knotx/junit5/util/RequestUtil.java

```diff
@@ -35,10 +38,16 @@ private RequestUtil() {}
   public static <T> void subscribeToResult_shouldFail(
       VertxTestContext context, Single<T> result, Consumer<Throwable> onError) {
     result
-        .doOnError(onError)
         .subscribe(
-            response -> context.failNow(new IllegalStateException("Error should occur")),
-            error -> context.completeNow());
+            response -> context.failNow(new AssertionError("Error should occur")),
+            error -> {
+              try {
+                onError.accept(error);
+                context.completeNow();
+              } catch (Throwable t) {
```

Comment:
Why `Throwable`?

Reply:
We need to handle both `AssertionError` and simple `Exceptions` here.

Reply:
I think the more specific version is better. We probably wouldn't want to catch some critical memory error

Reply:
Sure, agree. Fixed.